### PR TITLE
Introduce and structure label settings and text formats page

### PR DESCRIPTION
**Pull request by DelazJ**
Here's how the new TOC would look like ([current page](https://docs.qgis.org/testing/en/docs/user_manual/style_library/label_settings.html))
image
This allows to directly link the "text settings" dialog help button to the "formatting the label's text" section.
I'm just not happy with the "placement tab" and "rendering tab" titles. Keeping "tab" shows that they are at the same level than text, buffer... (in the label settings dialog) but it's less inline with the first same level title...
/me still overthinking...

Ticket: #

- [ ] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/user_manual/style_library/label_settings.rst [hunk 1]
```diff
@@ -13,18 +13,51 @@
    .. contents::
       :local:
 
+Labels are textual information you can display on vector features. They add
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
Labels are textual information you can display with vector features. They add
```

Comment (havatv) [reply source/docs/user_manual/style_library/label_settings.rst]: ?

File: source/docs/user_manual/style_library/label_settings.rst [hunk 2]
```diff
@@ -13,18 +13,51 @@
...
+details you could not necessarily represent using symbols.
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
information that is difficult to represent using symbols.
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 3]
```diff
@@ -13,18 +13,51 @@
...
 
-The following steps describe simple labeling without using the
-:guilabel:`Data defined override` functions, which are situated next to
-the drop-down menus - see :ref:`data_defined_labeling` for a use case.
+The :guilabel:`Style Manager` dialog allows you to create a set of labels or
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
The :guilabel:`Style Manager` dialog allows you to create a set of formatting styles for labels
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 4]
```diff
@@ -13,18 +13,51 @@
...
+text formats (ie the appearance of the text, including font, size, colors,
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
(ie the appearance of the text, including font, size, colors,
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 5]
```diff
@@ -13,18 +13,51 @@
...
+shadow, background...). Each of these items could later be applied to layers in
+the |labeling| :guilabel:`Labels` tab of the vector :guilabel:`Layer Properties`
+dialog or :guilabel:`Layer Styling` panel or using the |labeling| :sup:`Layer
+Labeling Options` button of the **Labels toolbar**. You can also directly
+configure them in the abovementioned dialogs.
 
+.. It could be nice to briefly expose the "label settings" tab of the style
+ manager, with some sample of label settings in it.
 
+The :guilabel:`Label Settings` dialog provides you with all the needed and
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
The :guilabel:`Label Settings` dialog allows you to
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 6]
```diff
@@ -13,18 +13,51 @@
...
+appropriate capabilities to configure smart labeling on vector layers. Setting
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
to configure smart labeling for vector layers. Setting
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 7]
```diff
@@ -13,18 +13,51 @@
...
+a label includes configuring the :ref:`text format <text_format>`, and how
+the label relates with the features or other labels (through :ref:`placement
+<labels_placement>`, :ref:`rendering <labels_rendering>` and callout).
+
+.. _text_format:
+
+Formatting the label's text
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
Formatting the label text
```

File: source/docs/user_manual/style_library/label_settings.rst [hunk 8]
```diff
@@ -13,18 +13,51 @@
...
+===========================
```
Comment (havatv) [inline source/docs/user_manual/style_library/label_settings.rst]:
```suggestion
=========================
```

File: source/docs/user_manual/style_library/style_manager.rst [hunk 9]
```diff
@@ -62,13 +62,13 @@ previewed items organized into tabs:
 * |lineLayer| :guilabel:`Line` for linear symbols only;
 * |polygonLayer| :guilabel:`Fill` for surface symbols only;
 * |color| :guilabel:`Color ramp`;
-* |text| :guilabel:`Text format` to manage text formats, which store the font,
-  color, buffers, shadows, and backgrounds of text formats (i.e. all the
-  formatting parts of label settings, which can be used outside of labels,
-  eg in layouts);
+* |text| :guilabel:`Text format` to manage :ref:`text formats <text_format>`,
+  which store the font, color, buffers, shadows, and backgrounds of text formats
+  (i.e. all the formatting parts of label settings, which can be used outside of
```
Comment (havatv) [inline source/docs/user_manual/style_library/style_manager.rst]:
```suggestion
  (i.e. all the formatting parts of the label settings, which for instance can be used
```

File: source/docs/user_manual/style_library/style_manager.rst [hunk 10]
```diff
@@ -62,13 +62,13 @@ previewed items organized into tabs:
...
+  labels, eg in layouts);
```
Comment (havatv) [inline source/docs/user_manual/style_library/style_manager.rst]:
```suggestion
  in layouts);
```

File: source/docs/user_manual/style_library/label_settings.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/user_manual/style_library/label_settings.rst]: This is not really the message I wanted to convey, ie from this dialog you create label settings AND text formats; 2 different items despite the latter is included in the former (which complicates description). the idea was also to introduce the "text formats" expression before it gets its own section.

Comment (havatv) [reply source/docs/user_manual/style_library/label_settings.rst]: OK. I was probably confused by the use of "or" (in "labels or text formats").

**Pull request merged**